code/datums/components/ghost_direct_control.dm

```diff
@@ -0,0 +1,143 @@
+	RegisterSignal(parent, COMSIG_MOB_LOGIN, PROC_REF(on_login))
+
+/datum/component/ghost_direct_control/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ATOM_ATTACK_GHOST, COMSIG_PARENT_EXAMINE, COMSIG_MOB_LOGIN))
+	return ..()
+
+/datum/component/ghost_direct_control/Destroy(force)
+	extra_control_checks = null
+	after_assumed_control = null
+	return ..()
+
+/// Inform ghosts that they can possess this
+/datum/component/ghost_direct_control/proc/on_examined(datum/source, mob/user, list/examine_text)
+	SIGNAL_HANDLER //COMSIG_PARENT_EXAMINE
+	if(!isobserver(user))
+		return
+	var/mob/living/our_mob = parent
+	if(our_mob.stat == DEAD || our_mob.key || awaiting_ghosts)
+		return
+	examine_text += "<span class='boldnotice'>You could take control of this mob by clicking on it.</span>"
+
+/// Send out a request for a brain
+/datum/component/ghost_direct_control/proc/request_ghost_control(poll_question, role_name, poll_length)
+	awaiting_ghosts = TRUE
+	var/list/candidates = SSghost_spawns.poll_candidates(
+		question = poll_question,
+		role = ban_type,
+		poll_time = poll_length,
+		source = parent,
+		role_cleanname = role_name,
+		flash_window = FALSE,
+		dont_play_notice_sound = TRUE
+	)
+	awaiting_ghosts = FALSE
+	var/mob/chosen_one = null
+	if(length(candidates))
+		chosen_one = pick(candidates)
+
+	if(isnull(chosen_one))
+		return
+	assume_direct_control(chosen_one)
+
+/// A ghost clicked on us, they want to get in this body
+/datum/component/ghost_direct_control/proc/on_ghost_clicked(mob/our_mob, mob/dead/observer/hopeful_ghost)
+	SIGNAL_HANDLER // COMSIG_ATOM_ATTACK_GHOST
+	if(our_mob.key)
+		qdel(src)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	if(!hopeful_ghost.client)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	if(awaiting_ghosts)
+		to_chat(hopeful_ghost, "<span class='warning'>Ghost candidate selection currently in progress!</span>")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	if(!SSticker.HasRoundStarted())
+		to_chat(hopeful_ghost, "<span class='warning'>You cannot assume control of this until after the round has started!</span>")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	INVOKE_ASYNC(src, PROC_REF(attempt_possession), our_mob, hopeful_ghost)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/// We got far enough to establish that this mob is a valid target, let's try to posssess it
+/datum/component/ghost_direct_control/proc/attempt_possession(mob/our_mob, mob/dead/observer/hopeful_ghost)
+	var/ghost_asked = tgui_alert(usr, "Become [our_mob]?", "Are you sure?", list("Yes", "No"))
+	if(ghost_asked != "Yes" || QDELETED(our_mob))
+		return
+	assume_direct_control(hopeful_ghost)
+
+/// Grant possession of our mob, component is now no longer required
+/datum/component/ghost_direct_control/proc/assume_direct_control(mob/harbinger)
+	if(QDELETED(src))
+		to_chat(harbinger, "<span class='warning'>Offer to possess creature has expired!</span>")
+		return
+	if(jobban_isbanned(harbinger, ban_type) || jobban_isbanned(harbinger, ROLE_SYNDICATE))
```

Comment:
This makes sense for its intended usecase, but it might be worth pulling the antag ban check up to a var on the type so this could be used for more than just heretic/cult stuff.

Reply:
Yeah let me do that

Reply:
Well, hmm.

We should always check for role_sentient

Reply:
done